### PR TITLE
[ie/yle_areena] Fix live events

### PR DESCRIPTION
**Pull request by CounterPillow**
Fixes extraction of manifests for live events.

**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

Previously, YLE live events like the [pupper stream](https://areena.yle.fi/1-72251830) would fail extraction with the following traceback:
```
  File "/usr/lib/python3.12/site-packages/yt_dlp/extractor/common.py", line 741, in extract
    ie_result = self._real_extract(url)
                ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/site-packages/yt_dlp/extractor/yle_areena.py", line 118, in _real_extract
    video_data['data']['ongoing_ondemand']['manifest_url'], video_id, 'mp4', m3u8_id='hls')
    ~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^
KeyError: 'ongoing_ondemand'
```

This change fixes it.

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### CODE REVIEW

File: yt_dlp/extractor/yle_areena.py [hunk 1]
```diff
@@ -114,8 +114,12 @@ def _real_extract(self, url):
                 'ie_key': KalturaIE.ie_key(),
             }
         else:
+            try:
+                ongoing = video_data['data']['ongoing_ondemand']
+            except KeyError:
+                ongoing = video_data['data']['ongoing_event']
```
Comment (bashonly) [inline yt_dlp/extractor/yle_areena.py:120]: I'd do this, and then also set `is_live = False` before the `if is_podcast:` line up above, as well as returning `'is_live': is_live` in the info dict below
```suggestion
            if traverse_obj(video_data, ('data', 'ongoing_ondemand', 'manifest_url', {url_or_none})):
                manifest_url = video_data['data']['ongoing_ondemand']['manifest_url']
            elif traverse_obj(video_data, ('data', 'ongoing_event', 'manifest_url', {url_or_none})):
                manifest_url = video_data['data']['ongoing_event']['manifest_url']
                is_live = True
            else:
                raise ExtractorError('Unable to extract manifest URL')
```

really the whole extractor could use a good refactoring (e.g. we don't access anything outside of the `'data'` branch in the `video_data` JSON, etc), but that's not necessary now

**Review by bashonly — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/yle_areena.py [hunk 2]
```diff
@@ -114,8 +114,12 @@ def _real_extract(self, url):
...
             formats, subs = self._extract_m3u8_formats_and_subtitles(
-                video_data['data']['ongoing_ondemand']['manifest_url'], video_id, 'mp4', m3u8_id='hls')
+                ongoing['manifest_url'], video_id, 'mp4', m3u8_id='hls')
```
Comment (bashonly) [inline yt_dlp/extractor/yle_areena.py:122]:
```suggestion
                'manifest_url', video_id, 'mp4', m3u8_id='hls', live=is_live)
```

**Comment by bashonly** [comment]
could you also add `https://areena.yle.fi/1-72251830` as a test?

**Comment by CounterPillow** [comment]
> could you also add `https://areena.yle.fi/1-72251830` as a test?

That would make very little sense as that test would break/change behaviour as soon as the stream is no longer live.

**Comment by bashonly** [comment]
You can mark it skip. I just want to have 1 livestream case in there. This is common among all extractors that support livestreams

**Comment by CounterPillow** [comment]
Sure, made your changes and added a test case, I didn't mark it skip though because all the other tests are also failing anyway so what's another one that'll be broken in a few days time.

**Review by bashonly — APPROVED** [review]

**Pull request merged**